apps/src/storage/dataBrowser/dataVisualizer/VisualizerModal.jsx

```diff
@@ -196,9 +198,39 @@ class VisualizerModal extends React.Component {
     return options.join(', ');
   }
 
+  countCells = memoize((records, tableColumns, chartType) => {
+    if (chartType !== ChartType.CROSS_TAB) {
+      return undefined;
+    } else {
+      return records.reduce((bucket, record) => {
+        tableColumns.forEach(column => {
+          bucket[column] ||= new Set();
+          bucket[column].add(record[column]);
+        });
+        return bucket;
+      }, new Set());
+    }
+  });
+
+  filterTableColumnsByCount = memoize(
```

Comment:
But as Ben mentioned already, we'd need to include a way to let users know why these columns are disabled.

Reply:
I did flip around the logic so that it'll keep the columns available, but disabled.

But I don't want to add in messages yet - we already don't display messages for non-numeric columns being disabled on other charts, so any messaging should be rolled into a bigger effort for consistent messaging across charts. Good follow up task!

Reply:
From a teaching perspective, I think:
- disabling non-numeric columns in histograms
- disabling columns that result in cell count greater than an arbitrarily set constant in cross-tab charts. 

are two different cases.

In the lesson plan, it is explicitly stated that 'Histograms can only be created with numerica data...'  [lesson_slides](https://docs.google.com/presentation/d/1jrF77jto1ZR_XWchlm0DHNxLygkx5L-FQQlqr9VYTU8/edit#slide=id.g2067e88de27_0_398) | [lesson_plan](https://studio.code.org/s/csp5-2023/lessons)

Screenshot of lesson plan:
![Screenshot 2024-01-18 at 12 24 30 PM](https://github.com/code-dot-org/code-dot-org/assets/107423305/a4e4da67-8887-4fbd-8c4e-3b535544138b)

But there isn't any mention in the lesson plan of limiting the columns based on size of dataset.

Maybe ask curriculum? I defer to them on this. Thanks!

Reply:
I'll file this as a future work item, there's too much UI consideration here to figure out how/when to display an error message, but I don't want to keep holding up this PR because it's still breaking in production.